### ISSUE TITLE
`onClickLaunch` stops working *at all* when exception is thrown inside

### ISSUE THREAD

**Issue by MikiLoz92** [rjaros/kvision#279]
Hi Rob, first of all thanks for all the effort you've put into KVision, we're about to launch (tomorrow actually!) our first production app with it and I've nothing to say but good things about this framework :)

I've had quite a really weird bug since I started working with KVision, and it's that every time I clicked a button that triggered some HTTP request, IF that request failed, it took me to the login screen of my app and every single button on the app stopped working.

Thought it must have to do with something in my logic but It wasn't until today that I started looking into and I think I figured the cause:

I was using `onClickLaunch` on all my buttons, and after careful inspection of that method I realized that `onClickLaunch` is using a custom `CoroutineScope`: `KVScope`. That seems to be the trouble, because if an exception is thrown inside the `onClickLaunch` method, the exception is propagated to the scope's default `Job` and it will fail, rendering that context inservible. 

I think the easiest solution to this would be to allow the user to provide a custom scope in the `onClickLaunch` function, I ended up making this extension and it appears to work:
```kotlin
inline fun <reified T : Widget> T.onClickLaunch(coroutineScope: CoroutineScope = CoroutineScope(window.asCoroutineDispatcher()) + SupervisorJob(), noinline handler: suspend T.(MouseEvent) -> Unit): Int {
    return this.setEventListener<T> {
        click = { e ->
            coroutineScope.launch {
                self.handler(e)
            }
        }
    }
}
```

If not, another solution might be putting a `SupervisorJob` instead of the default `defaultJob()`, like this:
```kotlin
val KVScope = CoroutineScope(window.asCoroutineDispatcher()) + SupervisorJob()
```

Good luck with KV5, btw!

**Comment by rjaros** [comment]
Thanks for your feedback!
You are absolutely right, the implementation should be better! I'll fix this in the next release.

Good luck with your app! :-)

**Comment by rjaros** [comment]
Released with KVision 5.0.0
